pkg/router/router_test.go

```diff
@@ -157,24 +163,101 @@ func TestAdmissionEdgeCases(t *testing.T) {
 	}
 }
 
+func TestConfigTemplateExecution(t *testing.T) {
+	// watching for errors
+	caughtErrors := []error{}
+	errCh, stopCh := make(chan error), make(chan struct{})
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func(ch chan error, stop chan struct{}, wg *sync.WaitGroup) {
+		defer wg.Done()
+	out:
+		for {
+			select {
+			case err := <-ch:
+				caughtErrors = append(caughtErrors, err)
+			case <-stop:
+				break out
```

Comment:
I think we can simplify the flow control and avoid using a label by using `return`:
```suggestion
		for {
			select {
			case err := <-ch:
				caughtErrors = append(caughtErrors, err)
			case <-stop:
				return
```
There is nothing after the `for` loop, and the deferred `wg.Done()` will still be executed when the `return` is reached.